### PR TITLE
C flavor

### PR DESCRIPTION
**Pull request by daltonv**
## Proposed Changes

Adds a new c_cpp flavor that can later be used by clang-format (see #763)

### Author/Contributor
- [x] Add entry to the [CHANGELOG](https://github.com/oxsecurity/megalinter/blob/main/CHANGELOG.md) listing the change and linking to the corresponding issue (if appropriate)
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Comment by daltonv** [comment]
I will still add clang-format. I just wanted to keep my first PR simple as this is my first contribution.

File: megalinter/descriptors/c.megalinter-descriptor.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ descriptor_id: C
 descriptor_type: language
 descriptor_flavors:
   - cupcake
+  - c_cpp
   - dotnet
```
Comment (daltonv) [inline megalinter/descriptors/c.megalinter-descriptor.yml:6]: I still see a problem here when it comes to `clang-format` being added. I would naturally put that linter in here and the cpp descriptor file.

However, if I understand it right, that would mean `clang-format` is also added to the `cupcake`, `dotnet`, and `dotnetweb` linters. I think we only want it in this new `c_cpp` linter. Right?

Comment (nvuillam) [reply megalinter/descriptors/c.megalinter-descriptor.yml:6]: @daltonv you can override descriptor_flavors at linter level, so you could add  the following property on clang-format linter definition :)

```
     descriptor_flavors: 
        - c_cpp
```

**Review by nvuillam — APPROVED** [review]
Looks very nice to me, many thanks for your contribution :)

**Pull request merged**